core/wiki/functions/search.tid

```diff
@@ -0,0 +1,4 @@
+title: $:/core/wiki/functions/search
+tags: $:/tags/Global
+
+\function tf.story-river.filter() [[list]addsuffix[<tv-story-list>]addsuffix[$(searchSuffix)$]substitute[]addsuffix[ ]addsuffix[:filter]addsuffix[$(searchPrefix)$]substitute[]addsuffix[{$:/state/story-river/filter}match]addsuffix[$(searchPrefix)$]substitute[]addsuffix[yes]addsuffix[$(searchSuffix)$]substitute[]addsuffix[then<currentTiddler>]addsuffix[search:]addsuffix{$:/config/StoryFilter/fields}addsuffix[:]addsuffix{$:/config/StoryFilter/flags}addsuffix[{$:/temp/search/input}]addsuffix[else{$:/state/story-river/filter}!match]addsuffix[$(searchPrefix)$]substitute[]addsuffix[yes]addsuffix[$(searchSuffix)$]substitute[]addsuffix[then<currentTiddler>]] +[addprefix[$(searchPrefix)$]substitute[]addsuffix[$(searchSuffix)$]substitute[]]
```

Comment:
Filters can also have several lines. In between filter runs you can add new-lines, which would probably make hat filter a bit more readable. 

I think it also uses some variables, that are defined somewhere else. IMO every "helper" definitions the filter needs should be defined with that filter. 

I also think it needs to be written in a way that actually is understandable. As it is at the moment it is a complete mess. 

May be new-lines will help already.